SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp

```diff
@@ -0,0 +1,458 @@
+	{
+		Worker_Alpha_PlayerIdentityTokenResponseFuture_Get(PITFuture, nullptr, this, &USpatialConnectionManager::OnPlayerIdentityToken);
+	}
+}
+
+void USpatialConnectionManager::ConnectToReceptionist(uint32 PlayInEditorID)
+{
+#if WITH_EDITOR
+	SpatialGDKServices::InitWorkers(bConnectAsClient, PlayInEditorID, ReceptionistConfig.WorkerId);
+#endif
+
+	ReceptionistConfig.PreConnectInit(bConnectAsClient);
+
+	ConfigureConnection ConnectionConfig(ReceptionistConfig, bConnectAsClient);
+
+	Worker_ConnectionFuture* ConnectionFuture = Worker_ConnectAsync(
+		TCHAR_TO_UTF8(*ReceptionistConfig.GetReceptionistHost()), ReceptionistConfig.ReceptionistPort,
+		TCHAR_TO_UTF8(*ReceptionistConfig.WorkerId), &ConnectionConfig.Params);
+
+	FinishConnecting(ConnectionFuture);
+}
+
+void USpatialConnectionManager::ConnectToLocator(FLocatorConfig* InLocatorConfig)
+{
+	if (InLocatorConfig == nullptr)
+	{
+		UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Trying to connect to locator with invalid locator config"));
+		return;
+	}
+
+	InLocatorConfig->PreConnectInit(bConnectAsClient);
+
+	ConfigureConnection ConnectionConfig(*InLocatorConfig, bConnectAsClient);
+
+	FTCHARToUTF8 PlayerIdentityTokenCStr(*InLocatorConfig->PlayerIdentityToken);
+	FTCHARToUTF8 LoginTokenCStr(*InLocatorConfig->LoginToken);
+
+	Worker_LocatorParameters LocatorParams = {};
+	FString ProjectName;
+	FParse::Value(FCommandLine::Get(), TEXT("projectName"), ProjectName);
+	LocatorParams.project_name = TCHAR_TO_UTF8(*ProjectName);
+	LocatorParams.credentials_type = Worker_LocatorCredentialsTypes::WORKER_LOCATOR_PLAYER_IDENTITY_CREDENTIALS;
+	LocatorParams.player_identity.player_identity_token = PlayerIdentityTokenCStr.Get();
+	LocatorParams.player_identity.login_token = LoginTokenCStr.Get();
+
+	// Connect to the locator on the default port(0 will choose the default)
+	WorkerLocator = Worker_Locator_Create(TCHAR_TO_UTF8(*InLocatorConfig->LocatorHost), SpatialConstants::LOCATOR_PORT, &LocatorParams);
+
+	Worker_ConnectionFuture* ConnectionFuture = Worker_Locator_ConnectAsync(WorkerLocator, &ConnectionConfig.Params);
+
+	FinishConnecting(ConnectionFuture);
+}
+
+void USpatialConnectionManager::FinishConnecting(Worker_ConnectionFuture* ConnectionFuture)
+{
+	TWeakObjectPtr<USpatialConnectionManager> WeakSpatialConnectionManager(this);
+
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [ConnectionFuture, WeakSpatialConnectionManager]
+	{
+		Worker_Connection* NewCAPIWorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
+		Worker_ConnectionFuture_Destroy(ConnectionFuture);
+
+		AsyncTask(ENamedThreads::GameThread, [WeakSpatialConnectionManager, NewCAPIWorkerConnection]
+		{
+			USpatialConnectionManager* SpatialConnectionManager = WeakSpatialConnectionManager.Get();
+
+			if (Worker_Connection_IsConnected(NewCAPIWorkerConnection))
+			{
+				SpatialConnectionManager->WorkerConnection = NewObject<USpatialWorkerConnection>();
```

Comment:
The next 10 lines are the only meaningful change from the original connection code in SpatialWorkerConnection.cpp.